frontend/src/app/[locale]/saved-grants/page.tsx

```diff
@@ -17,9 +22,56 @@ export async function generateMetadata({ params }: LocalizedPageProps) {
   return meta;
 }
 
+const SavedOpportunitiesList = ({
+  opportunities,
+}: {
+  opportunities: Opportunity[];
+}) => {
+  return (
+    <ul className="usa-prose usa-list--unstyled">
+      {opportunities.map((opportunity) => (
+        <li key={opportunity?.opportunity_id}>
+          <SearchResultsListItem opportunity={opportunity} saved={true} />
+        </li>
+      ))}
+    </ul>
+  );
+};
+
+const NoSavedOpportunities = ({
+  noSavedCTA,
+  searchButtonText,
+}: {
+  noSavedCTA: React.ReactNode;
+  searchButtonText: string;
+}) => {
+  return (
+    <>
+      <USWDSIcon
+        name="star_outline"
+        className="grid-col-1 usa-icon usa-icon--size-6 margin-top-4"
+      />
+      <div className="margin-top-2 grid-col-11">
+        <p className="usa-intro ">{noSavedCTA}</p>{" "}
+        <Link href="/search">
+          <Button type="button">{searchButtonText}</Button>
+        </Link>
+      </div>
+    </>
+  );
+};
+
 export default async function SavedGrants({ params }: LocalizedPageProps) {
   const { locale } = await params;
   const t = await getTranslations({ locale });
+  const savedOpportunities = await fetchSavedOpportunities();
```

Comment:
[a note for later] is there a way we could move this complexity to the API side so that we don't have to make so many round trips here, or will caching make that more or less a moot point?